pandas/tests/util/test_hashing.py

```diff
@@ -10,272 +10,320 @@
-        length = 2**(l_exp + 8) + l_add
-        s = tm.rands_array(length, 2)
-        result = hash_array(s, 'utf8')
-        assert not result[0] == result[1]
-
-    def test_hash_collisions(self):
-
-        # hash collisions are bad
-        # https://github.com/pandas-dev/pandas/issues/14711#issuecomment-264885726
-        L = ['Ingrid-9Z9fKIZmkO7i7Cn51Li34pJm44fgX6DYGBNj3VPlOH50m7HnBlPxfIwFMrcNJNMP6PSgLmwWnInciMWrCSAlLEvt7JkJl4IxiMrVbXSa8ZQoVaq5xoQPjltuJEfwdNlO6jo8qRRHvD8sBEBMQASrRa6TsdaPTPCBo3nwIBpE7YzzmyH0vMBhjQZLx1aCT7faSEx7PgFxQhHdKFWROcysamgy9iVj8DO2Fmwg1NNl93rIAqC3mdqfrCxrzfvIY8aJdzin2cHVzy3QUJxZgHvtUtOLxoqnUHsYbNTeq0xcLXpTZEZCxD4PGubIuCNf32c33M7HFsnjWSEjE2yVdWKhmSVodyF8hFYVmhYnMCztQnJrt3O8ZvVRXd5IKwlLexiSp4h888w7SzAIcKgc3g5XQJf6MlSMftDXm9lIsE1mJNiJEv6uY6pgvC3fUPhatlR5JPpVAHNSbSEE73MBzJrhCAbOLXQumyOXigZuPoME7QgJcBalliQol7YZ9',  # noqa
-             'Tim-b9MddTxOWW2AT1Py6vtVbZwGAmYCjbp89p8mxsiFoVX4FyDOF3wFiAkyQTUgwg9sVqVYOZo09Dh1AzhFHbgij52ylF0SEwgzjzHH8TGY8Lypart4p4onnDoDvVMBa0kdthVGKl6K0BDVGzyOXPXKpmnMF1H6rJzqHJ0HywfwS4XYpVwlAkoeNsiicHkJUFdUAhG229INzvIAiJuAHeJDUoyO4DCBqtoZ5TDend6TK7Y914yHlfH3g1WZu5LksKv68VQHJriWFYusW5e6ZZ6dKaMjTwEGuRgdT66iU5nqWTHRH8WSzpXoCFwGcTOwyuqPSe0fTe21DVtJn1FKj9F9nEnR9xOvJUO7E0piCIF4Ad9yAIDY4DBimpsTfKXCu1vdHpKYerzbndfuFe5AhfMduLYZJi5iAw8qKSwR5h86ttXV0Mc0QmXz8dsRvDgxjXSmupPxBggdlqUlC828hXiTPD7am0yETBV0F3bEtvPiNJfremszcV8NcqAoARMe']  # noqa
-
-        # these should be different!
-        result1 = hash_array(np.asarray(L[0:1], dtype=object), 'utf8')
-        expected1 = np.array([14963968704024874985], dtype=np.uint64)
-        tm.assert_numpy_array_equal(result1, expected1)
-
-        result2 = hash_array(np.asarray(L[1:2], dtype=object), 'utf8')
-        expected2 = np.array([16428432627716348016], dtype=np.uint64)
-        tm.assert_numpy_array_equal(result2, expected2)
-
-        result = hash_array(np.asarray(L, dtype=object), 'utf8')
-        tm.assert_numpy_array_equal(
-            result, np.concatenate([expected1, expected2], axis=0))
+@pytest.fixture(params=[
+    Series([1, 2, 3] * 3, dtype="int32"),
+    Series([None, 2.5, 3.5] * 3, dtype="float32"),
+    Series(["a", "b", "c"] * 3, dtype="category"),
+    Series(["d", "e", "f"] * 3),
+    Series([True, False, True] * 3),
+    Series(pd.date_range("20130101", periods=9)),
+    Series(pd.date_range("20130101", periods=9, tz="US/Eastern")),
+    Series(pd.timedelta_range("2000", periods=9))])
+def series(request):
+    return request.param
+
+
+def _check_equal(obj, **kwargs):
+    """
+    Check that hashing an objects produces the same value each time.
+
+    Parameters
+    ----------
+    obj : object
+        The object to hash.
+    kwargs : kwargs
+        Keyword arguments to pass to the hashing function.
+    """
+    a = hash_pandas_object(obj, **kwargs)
+    b = hash_pandas_object(obj, **kwargs)
+    tm.assert_series_equal(a, b)
+
+    kwargs.pop("index", None)
+
+    a = hash_pandas_object(obj, **kwargs)
```

Comment:
Hmm I know this is how it was before, but this looks buggy. We don't seem to ever call `_check_equal` with an `index=` kwarg, so we always use the default of `index=True`.

So I think we're testing that `hash_pandas_object(obj, index=True)` twice, and not checking that `hash_pandas_object(obj, index=False)` at all. Or am I missing something?

Reply:
Nope, you're not missing anything.  Let's fix that.